cabal.project.bootstrap

```diff
@@ -0,0 +1,12 @@
+packages:
+    Cabal
+  , Cabal-syntax
+  , Cabal-hooks
+  , cabal-install
+  , cabal-install-solver
+
+-- Don't include tests or benchmarks for bootstrapping
+tests: False
+benchmarks: False
+
+index-state: hackage.haskell.org 2024-04-22T06:16:57Z

```

Comment:
Proliferation of `index-states` is concerning. Ideally, it'd be done with `import`'s.

Reply:
I think #9901 makes that difficult currently.

Reply:
Yeah, I haven't put enough thought into possible workarounds. I guess, this PR should be merged asap, and we can think about it more, later.